dist/src/main/java/io/camunda/application/commons/console/ping/PingConsoleConfiguration.java

```
@@ -0,0 +1,114 @@
+/*
+ * Copyright Camunda Services GmbH and/or licensed to Camunda Services GmbH under
+ * one or more contributor license agreements. See the NOTICE file distributed
+ * with this work for additional information regarding copyright ownership.
+ * Licensed under the Camunda License 1.0. You may not use this file
+ * except in compliance with the Camunda License 1.0.
+ */
+package io.camunda.application.commons.console.ping;
+
+import io.camunda.application.commons.console.ping.PingConsoleConfiguration.ConsolePingConfiguration;
+import io.camunda.service.ManagementServices;
+import io.camunda.zeebe.util.error.FatalErrorHandler;
+import java.util.Map;
+import java.util.concurrent.ScheduledThreadPoolExecutor;
+import java.util.concurrent.TimeUnit;
+import org.slf4j.Logger;
+import org.slf4j.LoggerFactory;
+import org.springframework.beans.factory.annotation.Autowired;
+import org.springframework.boot.ApplicationArguments;
+import org.springframework.boot.ApplicationRunner;
+import org.springframework.boot.context.properties.ConfigurationProperties;
+import org.springframework.boot.context.properties.EnableConfigurationProperties;
+import org.springframework.context.annotation.Configuration;
+
+@Configuration
+@EnableConfigurationProperties({ConsolePingConfiguration.class})
+public class PingConsoleConfiguration implements ApplicationRunner {
```

:question: Why is this an application runner? It will run as a separate application from the main one?

---

Yes, thats currently the implementation, when if fails the vaildation it stops the whole application at the start. Should we still run the app if this fails?

---

For now let's assume yes, this feature should be unintrusive. We can get product to decide if we want to fail early on misconfiguration.